src/addresses/address-details/AddressDetails.container.tsx

```diff
@@ -0,0 +1,133 @@
+import { FC, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { AddressDetailsBreadcrumb } from '../../shared-components/AddressDetailsBreadcrumb/AddressDetailsBreadcrumb';
+import {
+  PageSection,
+  PageSectionVariants,
+  Title,
+} from '@patternfly/react-core';
+import { useTranslation } from '../../i18n';
+import {
+  AMQBrokerModel,
+  AuthContext,
+  getApiServerBaseUrl,
+  useJolokiaLogin,
+} from '../../utils';
+import {
+  K8sResourceKind,
+  k8sGet,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { OpenAPI as OpenAPIConfig } from '../../openapi/jolokia/requests/core/OpenAPI';
+import { AddressDetails } from './AddressDetails.component';
+
+const AddressDetailsPage: FC = () => {
+  const { t } = useTranslation();
+  const {
+    name,
+    ns: namespace,
+    brokerName,
+    podName,
+  } = useParams<{
+    name?: string;
+    ns?: string;
+    brokerName?: string;
+    podName?: string;
+  }>();
+  const [brokerDetails, setBrokerDetails] = useState<K8sResourceKind>({});
+  const [_loading, setLoading] = useState<boolean>(true);
+  const [routes] = useK8sWatchResource<K8sResourceKind[]>({
+    isList: true,
+    groupVersionKind: {
+      group: 'route.openshift.io',
+      kind: 'Route',
+      version: 'v1',
+    },
+    namespaced: true,
+  });
+
+  const k8sGetBroker = () => {
+    setLoading(true);
+    k8sGet({ model: AMQBrokerModel, name: brokerName, ns: namespace })
+      .then((broker: K8sResourceKind) => {
+        setBrokerDetails(broker);
+      })
+      .catch((e) => {
+        console.error(e);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  };
+
+  useEffect(() => {
+    k8sGetBroker();
+  }, []);
+
+  const podOrdinal = parseInt(podName.replace(brokerName + '-ss-', ''));
+
+  const { token, isSucces, isLoading, source } = useJolokiaLogin(
+    brokerDetails,
+    routes,
+    podOrdinal,
+  );
+
+  const [prevIsLoading, setPrevIsLoading] = useState(isLoading);
+  const [notify, setNotify] = useState(false);
+  if (prevIsLoading !== isLoading) {
+    if (!isLoading && source === 'api') {
+      setNotify(true);
+    }
+    setPrevIsLoading(isLoading);
+  }
+  if (notify) {
+    if (isSucces) {
+      // TODO maybe use the OpenShift console notification system to let the
+      // user know that the login was a success?
+      alert(
+        'login successful ' + brokerDetails?.metadata?.name + ' token ' + token,
+      );
+    } else {
+      alert('login failed');
+    }
+    setNotify(false);
+  }
```

Comment:
We need to figure a way to notify the user without alerts, but that's out of the scope of this PR.

Reply:
yes.

Reply:
There would be a login solution when we have a good idea how to to authentication/authorization.